cmd/botkube/main.go

```diff
@@ -74,11 +74,11 @@ func run(ctx context.Context) error {
 	// Load configuration
 	intconfig.RegisterFlags(pflag.CommandLine)
 
-	remoteCfgSyncEnabled := graphql.IsRemoteConfigEnabled()
-	gqlClient := graphql.NewDefaultGqlClient()
-	deployClient := intconfig.NewDeploymentClient(gqlClient)
+	remoteCfg, remoteCfgEnabled := remote.GetConfig()
```

Comment:
we need to finally clean up this main func. We are creating real clients just to not use them in case of `remoteCfgEnabled`. 

With such pattern we will in a moment end-up with creating no-op implementation that we will pass to other "getter" to return no-op with no-op deps 🙈

Reply:
True :/ I tried to improve things a bit, but I know it's not enough. Will sync with you tomorrow 👍

Reply:
Ok, ensured that we don't create clients when remote config is enabled:

```go
	remoteCfg, remoteCfgEnabled := remote.GetConfig()
	var (
		gqlClient    *remote.Gql
		deployClient *remote.DeploymentClient
	)
	if remoteCfgEnabled {
		gqlClient = remote.NewDefaultGqlClient(remoteCfg)
		deployClient = remote.NewDeploymentClient(gqlClient)
	}
```

Let's follow-up on this in https://github.com/kubeshop/botkube/issues/745, as this is IMO quite connected